src/popup/component/AccountIcon/index.js

```diff
@@ -13,9 +14,14 @@ export default class AccountIcon extends PureComponent {
         this.iconContainer.current.appendChild(image)
     }
     generateJazziconSvg = (address, diameter) => {
-        const numericRepresentation = this.addressToNumber(address)
+        const numericRepresentation = isEvmAddress(address) ? this.jsNumberForAddress(address):this.addressToNumber(address)
```

Comment:
here `address` is always the oasis address. we'd need to pass in the evmAddress from outside to use jsNumberForAddress.

<details>

```diff
diff --git a/src/popup/component/AccountIcon/index.js b/src/popup/component/AccountIcon/index.js
index 113b7d2..6503bae 100644
--- a/src/popup/component/AccountIcon/index.js
+++ b/src/popup/component/AccountIcon/index.js
@@ -9,12 +9,12 @@ export default class AccountIcon extends PureComponent {
         this.appendJazzicon()
     }
     appendJazzicon() {
-        const { address, diameter } = this.props
-        const image = this.generateJazziconSvg(address, diameter)
+        const { address, evmAddress, diameter } = this.props
+        const image = this.generateJazziconSvg(address, evmAddress, diameter)
         this.iconContainer.current.appendChild(image)
     }
-    generateJazziconSvg = (address, diameter) => {
-        const numericRepresentation = isEvmAddress(address) ? this.jsNumberForAddress(address):this.addressToNumber(address)
+    generateJazziconSvg = (address, evmAddress, diameter) => {
+        const numericRepresentation = evmAddress ? this.jsNumberForAddress(evmAddress):this.addressToNumber(address)
         return jazzicon(diameter, numericRepresentation)
     }
     jsNumberForAddress(address) {
diff --git a/src/popup/component/WalletBar/index.js b/src/popup/component/WalletBar/index.js
index d226695..f0d35f2 100644
--- a/src/popup/component/WalletBar/index.js
+++ b/src/popup/component/WalletBar/index.js
@@ -110,9 +110,11 @@ class WalletBar extends React.Component {
     renderMyIcon = () => {
         let { currentAccount } = this.props
         let address = currentAccount.address
+        let evmAddress = currentAccount.evmAddress
         return (
             <AccountIcon
                 address={address}
+                evmAddress={evmAddress}
                 diameter={"30"}
             />
         )
```

</details>

Reply:
oh there are some other places that use `<AccountIcon>` that also would need to be updated

Reply:
Here is the incoming evmAddress from the outside. Let me fix it

Reply:
i fix evm pass in[ there ](https://github.com/oasisprotocol/oasis-wallet-ext/pull/168/commits/ec72bb02cc866b69448f5cfb94c133b1ee406fd4)

Reply:
confirmed, thanks!